dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/acl/DefaultTrackerOwnershipManager.java

```diff
@@ -178,51 +199,29 @@ public void grantTemporaryOwnership(
         getTempOwnershipCacheKey(trackedEntity.getUid(), program.getUid(), user.getUid()));
   }
 
-  private void validateGrantTemporaryOwnershipInputs(
-      TrackedEntity trackedEntity, Program program, UserDetails user) throws ForbiddenException {
-    if (program == null) {
-      throw new ForbiddenException(
-          "Temporary ownership not created. Program supplied does not exist.");
-    }
-
+  private void validateTrackedEntity(TrackedEntity trackedEntity, UserDetails user)
+      throws ForbiddenException {
     if (trackedEntity == null) {
       throw new ForbiddenException(
```

Comment:
If we use the `TrackedEntityService` we do not need to worry about this but this one should be more a `NOT_FOUND` exception

Reply:
as explained [here](https://github.com/dhis2/dhis2-core/pull/20637/files#r2046343406), I can't use the `TrackedEntityService`, but I agree with throwing a `NOT_FOUND`